astropy/wcs/wcs.py

```diff
@@ -2919,6 +2919,9 @@ def slice(self, view, numpy_order=True):
 
         wcs_new = self.deepcopy()
         for i, iview in enumerate(view):
+            if iview.step < 0:
```

Comment:
@keflavich - I think you probably want something like ``if iview.step is not None and iview.step < 0``